internal/server/ofrep/errors.go

```diff
@@ -0,0 +1,35 @@
+package ofrep
+
+import (
+	"encoding/json"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type errorCode string
+
+const (
+	errorCodeFlagNotFound        errorCode = "FLAG_NOT_FOUND"
+	errorCodeParseError          errorCode = "PARSE_ERROR"
+	errorCodeTargetingKeyMissing errorCode = "TARGETING_KEY_MISSING"
+	errorCodeInvalidContext      errorCode = "INVALID_CONTEXT"
+	errorCodeParseGeneral        errorCode = "GENERAL"
+)
+
+type errorSchema struct {
+	Key          string `json:"key,omitempty"`
+	ErrorCode    string `json:"errorCode,omitempty"`
+	ErrorDetails string `json:"errorDetails"`
+}
+
+func NewNotFoundError(key string) error {
+	msg, err := json.Marshal(&errorSchema{
+		Key:       key,
+		ErrorCode: string(errorCodeFlagNotFound),
+	})
+	if err != nil {
+		panic(err)
```

Comment:
prob dont want to panic here, just return the error as status `INTERNAL` code for grpc with the error msg